.github/workflows/run-dev-tests.yml

```diff
@@ -0,0 +1,114 @@
+name: "Check python code"
+
+on:
+  push:
+    paths:
+      - "**.py"
+      - .pylintrc
+      - mypy.ini
```

Comment:
mypy.ini seems to have been merged into `pyproject.toml`

Reply:
oh yea. I'll change this to the pyproject.toml and pre-commit.yml

I consolidated linter configs into the toml to avoid repo clutter. I want is people to see the README when they browse the root folder on the website. If there are too many folder/files in the root folder then they have to scroll down to see the README... It looks like I can hide the [coverage config in the toml](https://coverage.readthedocs.io/en/latest/config.html) also.

Reply:
I updated these triggers.

I moved the coverage config to the toml. I also added a basic pytest config to the toml, so it doesn't need any args passed to the `pytest` command.

Reply:
With all the new hooks added to pre-commit, it might be better to just run this workflow regardless of the paths. Currently, the yaml and requirements.txt files aren't getting checked in the CI unless there was also changes made to
https://github.com/cpp-linter/cpp-linter-action/blob/23622e1efccbff34aa4ab0b299bef38deb127a5f/.github/workflows/run-dev-tests.yml#L5-L9

Reply:
Nevermind, I just moved the pre-commit steps into its own CI workflow.